Framework/CurveFitting/inc/MantidCurveFitting/Algorithms/PlotPeakByLogValueHelper.h

```diff
@@ -0,0 +1,51 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2008 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#ifndef MANTID_CURVEFITTING_PLOTPEAKBULOGVALUEHELPER_H_
+#define MANTID_CURVEFITTING_PLOTPEAKBULOGVALUEHELPER_H_
+#include "MantidAPI/DllConfig.h"
+
+#include "MantidAPI/AlgorithmManager.h"
+#include "MantidAPI/IAlgorithm_fwd.h"
+#include "MantidAPI/MatrixWorkspace_fwd.h"
+#include "MantidAPI/Workspace_fwd.h"
+
+#include <boost/optional.hpp>
+#include <string>
+#include <vector>
+
+namespace Mantid {
+namespace CurveFitting {
+namespace Algorithms {
+
+struct InputData {
+  /// Constructor
+  InputData(const std::string &nam, int ix, int p)
+      : name(nam), i(ix), period(p) {}
+  /// Copy constructor
+  InputData(const InputData &data)
+      : name(data.name), i(data.i), period(data.period), ws(data.ws) {}
+  std::string name; ///< Name of a workspace or file
+  int i;            ///< Workspace index of the spectra to fit
+  int period;       ///< Period, needed if a file contains several periods
+  API::MatrixWorkspace_sptr ws; ///< shared pointer to the workspace
```

Comment:
Are all InputData objects guaranteed to be in scope as long as the algorithm? (I.e. all InputData objects belong to classes which hold the sptr?)

If so we should make this a `API::MatrixWorkspace &`

Reply:
The InputData objects own the pointer in this case.